### PR TITLE
Feat/condo/doma 8371/support state organisations

### PR DESCRIPTION
**Pull request by abshnko**
Supports routingNumbers (BIK) that start with '01' (used by state funded organisations). Added correct validations. 

Adds a new optional field `classificationCode` (KBK) to `BankAccount` and `BillingRecipient` models
`classificationCode` is a special code that is used by organisations to indicate the purpose of government funds. 

Refactor: used GQLError instead of validationFailureErorr in `BankAccount` model for K6 support

### CODE REVIEW

File: apps/condo/domains/banking/utils/validate/countrySpecificValidators/ru.validator.js [hunk 1]
```diff
@@ -37,17 +40,17 @@ const validateRuNumber = (number, routingNumber, errors) => {
...
  * *** In Russia, number routing is equivalent to BIC***
- * 1) For RU organizations country code (first two digits) is 04
+ * 1) For RU organizations country code (first two digits) is 04 or 01 (for government organizations)
  *
  * @param {string} routingNumber
  * @param {[]} errors
  */
 const validateRuRoutingNumber = (routingNumber, errors) => {
-    const WRONG_RU_COUNTRY_CODE = 'For RU organizations country code is 04, but routing number have '
+    const WRONG_RU_COUNTRY_CODE = 'For RU organizations country code is 04 or 01, but routing number has '
```
Comment (toplenboren) [inline apps/condo/domains/banking/utils/validate/countrySpecificValidators/ru.validator.js:49]: Seems that this sentence ends abruptly

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/banking/schema/BankAccount.js [hunk 2]
```diff
@@ -68,7 +89,7 @@ const BankAccount = new GQLListSchema('BankAccount', {
                     const { result, errors } = validateTin(tin, country)
 
                     if ( !result ) {
-                        addFieldValidationError(errors[0])
+                        throw new GQLError(getError(TIN_IS_INVALID, errors[0], 'classificationCode'), context)
```
Comment (AleX83Xpert) [inline apps/condo/domains/banking/schema/BankAccount.js:92]: The error is about the `tin`, but field is `classificationCode`. Is it ok?

File: apps/condo/domains/banking/schema/BankAccount.js [hunk 3]
```diff
@@ -101,7 +122,7 @@ const BankAccount = new GQLListSchema('BankAccount', {
                     const { result, errors } = validateRoutingNumber(routingNumber, country)
 
                     if ( !result ) {
-                        addFieldValidationError(errors[0])
+                        throw new GQLError(getError(ROUTING_NUMBER_IS_INVALID, errors[0], 'classificationCode'), context)
```
Comment (AleX83Xpert) [inline apps/condo/domains/banking/schema/BankAccount.js:125]: routing number VS classification code :)

File: apps/condo/domains/banking/schema/BankAccount.js [hunk 4]
```diff
@@ -128,7 +149,7 @@ const BankAccount = new GQLListSchema('BankAccount', {
                     const { result, errors } = validateNumber(number, routingNumber, country)
 
                     if ( !result ) {
-                        addFieldValidationError(errors[0])
+                        throw new GQLError(getError(NUMBER_IS_INVALID, errors[0], 'classificationCode'), context)
```
Comment (AleX83Xpert) [inline apps/condo/domains/banking/schema/BankAccount.js:152]: "Number" or "classificationCode"

File: apps/condo/domains/banking/schema/BankAccount.test.js [hunk 5]
```diff
@@ -570,11 +569,16 @@ describe('BankAccount', () => {
...
+
+            await expectToThrowGQLError(async () => {
                 await createTestBankAccount(adminClient, organization, {
                     integrationContext: { connect: { id: anotherBankAccount.integrationContext.id } },
                 })
-            }, `Cannot connect to BankIntegrationAccountContext, used by another BankAccount(id="${anotherBankAccount.id}")`)
+            }, {
+                code: 'BAD_USER_INPUT',
+                type: 'BANK_INTEGRATION_ACCOUNT_CONTEXT_ALREADY_USED',
+                message: `Cannot connect to BankIntegrationAccountContext, used by another BankAccount(id="${anotherBankAccount.id}")`,
```
Comment (AleX83Xpert) [inline apps/condo/domains/banking/schema/BankAccount.test.js:580]: It's also good to check the field if possible. Here and below.

File: apps/condo/domains/banking/schema/BankAccount.js [hunk 6]
```diff
@@ -179,6 +200,26 @@ const BankAccount = new GQLListSchema('BankAccount', {
             type: Json,
             isRequired: false,
         },
+
+        classificationCode: {
+            schemaDoc: 'Budget classification code, used for state-funded organizations',
+            type: Text,
+            isRequired: false,
+            hooks: {
+                validateInput: ({ context, existingItem, resolvedData }) => {
```
Comment (AleX83Xpert) [inline apps/condo/domains/banking/schema/BankAccount.js:209]: Any test for this validation?

File: apps/condo/domains/banking/utils/validate/countrySpecificValidators/ru.validator.js (hunk 1 above)
Comment (abshnko) [reply apps/condo/domains/banking/utils/validate/countrySpecificValidators/ru.validator.js:49]: it uses `routingNumber` variable at the end (to compare what is expected vs what is actually there)

File: apps/condo/domains/banking/schema/BankAccount.js (hunk 6 above)
Comment (abshnko) [reply apps/condo/domains/banking/schema/BankAccount.js:209]: yes, in validation test itself (apps/condo/domains/banking/utils/validate/classificationCode.utils.spec.js)

**Review by dkoviazin — APPROVED** [review]

**Review by AleX83Xpert — APPROVED** [review]

**Review by VKislov — APPROVED** [review]

**Pull request merged**